src/core/thread/mle_router.cpp

```diff
@@ -1359,11 +1351,11 @@ ThreadError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::M
             ExitNow();
         }
 
-        if (memcmp(&mParent.mMacAddr, &macAddr, sizeof(mParent.mMacAddr)) == 0)
+        if (memcmp(&mParent.GetExtAddress(), &macAddr, sizeof(mParent.GetExtAddress())) == 0)
```

Comment:
That operator is not yet defined, was planning to do that general cleanup a separate PR.